examples/chat/listener.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+/* eslint-disable no-console */
+
+const PeerId = require('peer-id')
+const PeerInfo = require('peer-info')
+const libp2p = require('../../src')
+const multiaddr = require('multiaddr')
+const pull = require('pull-stream')
+
+PeerId.createFromJSON(require('./peer-id-listener'), (err, idListener) => {
+  if (err) {
+    throw err
+  }
+  const peerListener = new PeerInfo(idListener)
+  peerListener.multiaddr.add(multiaddr('/ip4/0.0.0.0/tcp/10333'))
+  const nodeListener = new libp2p.Node(peerListener)
+
+  nodeListener.start((err) => {
+    if (err) {
+      throw err
+    }
+
+    nodeListener.swarm.on('peer-mux-established', (peerInfo) => {
+      console.log(peerInfo.id.toB58String())
+    })
+
+    nodeListener.handle('/chat/1.0.0', (protocol, conn) => {
+      pull(
+        conn,
+        pull.collect((err, data) => {
```

Comment:
you probably want to use https://pull-stream.github.io/#drain otherwise this function will be only called when the other stream is over